# Расчет воронки за 4 месяца относительно пола пользователя и типа устройства

## 1 Расчет воронок

### 1.1 Пользователь: Мужчина, устройство: Мобильный телефон

In [2]:
import csv

funnel_month_mobile_male = {}
funnel_template = {'1_home_page': 0, '2_search_page': 0, '3_payment_page': 0, '4_payment_confirmation_page': 0}
with open('click_stream3.csv', mode='r') as csv_file: # открываем файл
    csv_reader = csv.DictReader(csv_file, fieldnames=['ID','page','date', 'device', 'Gender']) # читаем файл
    
    for row in csv_reader:
        
        page = list(row.items())[1][1]
        device = list(row.items())[3][1]
        Gender = list(row.items())[4][1]
        
        event_date = list(row.items())[2][1][:-3]
        
        if event_date not in funnel_month_mobile_male:
            funnel_month_mobile_male[event_date] = funnel_template.copy()
            
        if device == 'Mobile' and Gender == 'Male':
            
            
            if page == '1_home_page':
                funnel_month_mobile_male[event_date]['1_home_page'] += 1
            elif page == '2_search_page':
                funnel_month_mobile_male[event_date]['2_search_page'] += 1
            elif page == '3_payment_page':
                funnel_month_mobile_male[event_date]['3_payment_page'] += 1
            elif page == '4_payment_confirmation_page':
                funnel_month_mobile_male[event_date]['4_payment_confirmation_page'] += 1

    print('Таблица 1\t', 'Воронка, пользователь: Мужчина, устройство: Мобильный телефон, кол-во переходов')
    print('Страница/Месяц\t\t\t', '2015-01\t', '2015-02\t', '2015-03\t', '2015-04')
    print('1_home_page\t\t\t', funnel_month_mobile_male['2015-01']['1_home_page'], '\t\t', funnel_month_mobile_male['2015-02']['1_home_page'], '\t\t', funnel_month_mobile_male['2015-03']['1_home_page'], '\t\t', funnel_month_mobile_male['2015-04']['1_home_page'])
    print('2_search_page\t\t\t', funnel_month_mobile_male['2015-01']['2_search_page'], '\t\t', funnel_month_mobile_male['2015-02']['2_search_page'], '\t\t', funnel_month_mobile_male['2015-03']['2_search_page'], '\t\t', funnel_month_mobile_male['2015-04']['2_search_page'])
    print('3_payment_page\t\t\t', funnel_month_mobile_male['2015-01']['3_payment_page'], '\t\t', funnel_month_mobile_male['2015-02']['3_payment_page'], '\t\t', funnel_month_mobile_male['2015-03']['3_payment_page'], '\t\t', funnel_month_mobile_male['2015-04']['3_payment_page'])
    print('4_payment_confirmation_page\t', funnel_month_mobile_male['2015-01']['4_payment_confirmation_page'], '\t\t', funnel_month_mobile_male['2015-02']['4_payment_confirmation_page'], '\t\t', funnel_month_mobile_male['2015-03']['4_payment_confirmation_page'], '\t\t', funnel_month_mobile_male['2015-04']['4_payment_confirmation_page'])
    print('\n')
   

Таблица 1	 Воронка, пользователь: Мужчина, устройство: Мобильный телефон, кол-во переходов
Страница/Месяц			 2015-01	 2015-02	 2015-03	 2015-04
1_home_page			 3682 		 3799 		 3812 		 3829
2_search_page			 2963 		 3035 		 762 		 755
3_payment_page			 551 		 591 		 160 		 148
4_payment_confirmation_page	 53 		 59 		 10 		 13




### 1.2 Пользователь: Женщина, устройство: Мобильный телефон

In [3]:

funnel_month_mobile_female = {}
funnel_template = {'1_home_page': 0, '2_search_page': 0, '3_payment_page': 0, '4_payment_confirmation_page': 0}
with open('click_stream3.csv', mode='r') as csv_file: # открываем файл
    csv_reader = csv.DictReader(csv_file, fieldnames=['ID','page','date', 'device', 'Gender']) # читаем файл
    
    for row in csv_reader:
        
        page = list(row.items())[1][1]
        device = list(row.items())[3][1]
        Gender = list(row.items())[4][1]
        event_date = list(row.items())[2][1][:-3]
        
        if event_date not in funnel_month_mobile_female:
            funnel_month_mobile_female[event_date] = funnel_template.copy()
            
        if device == 'Mobile' and Gender == 'Female':
            
            
            if page == '1_home_page':
                funnel_month_mobile_female[event_date]['1_home_page'] += 1
            elif page == '2_search_page':
                funnel_month_mobile_female[event_date]['2_search_page'] += 1
            elif page == '3_payment_page':
                funnel_month_mobile_female[event_date]['3_payment_page'] += 1
            elif page == '4_payment_confirmation_page':
                funnel_month_mobile_female[event_date]['4_payment_confirmation_page'] += 1

    print('Таблица 2\t', 'Воронка, пользователь: Женщина, устройство: Мобильный телефон, кол-во переходов')
    print('Страница/Месяц\t\t\t', '2015-01\t', '2015-02\t', '2015-03\t', '2015-04')
    print('1_home_page\t\t\t', funnel_month_mobile_female['2015-01']['1_home_page'], '\t\t', funnel_month_mobile_female['2015-02']['1_home_page'], '\t\t', funnel_month_mobile_female['2015-03']['1_home_page'], '\t\t', funnel_month_mobile_female['2015-04']['1_home_page'])
    print('2_search_page\t\t\t', funnel_month_mobile_female['2015-01']['2_search_page'], '\t\t', funnel_month_mobile_female['2015-02']['2_search_page'], '\t\t', funnel_month_mobile_female['2015-03']['2_search_page'], '\t\t', funnel_month_mobile_female['2015-04']['2_search_page'])
    print('3_payment_page\t\t\t', funnel_month_mobile_female['2015-01']['3_payment_page'], '\t\t', funnel_month_mobile_female['2015-02']['3_payment_page'], '\t\t', funnel_month_mobile_female['2015-03']['3_payment_page'], '\t\t', funnel_month_mobile_female['2015-04']['3_payment_page'])
    print('4_payment_confirmation_page\t', funnel_month_mobile_female['2015-01']['4_payment_confirmation_page'], '\t\t', funnel_month_mobile_female['2015-02']['4_payment_confirmation_page'], '\t\t', funnel_month_mobile_female['2015-03']['4_payment_confirmation_page'], '\t\t', funnel_month_mobile_female['2015-04']['4_payment_confirmation_page'])
    print('\n')
   

Таблица 2	 Воронка, пользователь: Женщина, устройство: Мобильный телефон, кол-во переходов
Страница/Месяц			 2015-01	 2015-02	 2015-03	 2015-04
1_home_page			 3868 		 3751 		 3738 		 3721
2_search_page			 3062 		 3020 		 747 		 756
3_payment_page			 618 		 630 		 175 		 147
4_payment_confirmation_page	 76 		 60 		 20 		 11




### 1.3 Пользователь: Мужчина, устройство: ПК

In [4]:
funnel_month_desktop_male = {}
funnel_template = {'1_home_page': 0, '2_search_page': 0, '3_payment_page': 0, '4_payment_confirmation_page': 0}
with open('click_stream3.csv', mode='r') as csv_file: # открываем файл
    csv_reader = csv.DictReader(csv_file, fieldnames=['ID','page','date', 'device', 'Gender']) # читаем файл
    
    for row in csv_reader:
        
        page = list(row.items())[1][1]
        device = list(row.items())[3][1]
        Gender = list(row.items())[4][1]
        event_date = list(row.items())[2][1][:-3]
        
        if event_date not in funnel_month_desktop_male:
            funnel_month_desktop_male[event_date] = funnel_template.copy()
            
        if  device == 'Desktop' and Gender == 'Male': 
            
            
            if page == '1_home_page':
                funnel_month_desktop_male[event_date]['1_home_page'] += 1
            elif page == '2_search_page':
                funnel_month_desktop_male[event_date]['2_search_page'] += 1
            elif page == '3_payment_page':
                funnel_month_desktop_male[event_date]['3_payment_page'] += 1
            elif page == '4_payment_confirmation_page':
                funnel_month_desktop_male[event_date]['4_payment_confirmation_page'] += 1

    print('Таблица 3\t', 'Воронка, пользователь: Мужчина, устройство: ПК, кол-во переходов')
    print('Страница/Месяц\t\t\t', '2015-01\t', '2015-02\t', '2015-03\t', '2015-04')
    print('1_home_page\t\t\t', funnel_month_desktop_male['2015-01']['1_home_page'], '\t\t', funnel_month_desktop_male['2015-02']['1_home_page'], '\t\t', funnel_month_desktop_male['2015-03']['1_home_page'], '\t\t', funnel_month_desktop_male['2015-04']['1_home_page'])
    print('2_search_page\t\t\t', funnel_month_desktop_male['2015-01']['2_search_page'], '\t\t', funnel_month_desktop_male['2015-02']['2_search_page'], '\t\t', funnel_month_desktop_male['2015-03']['2_search_page'], '\t\t', funnel_month_desktop_male['2015-04']['2_search_page'])
    print('3_payment_page\t\t\t', funnel_month_desktop_male['2015-01']['3_payment_page'], '\t\t', funnel_month_desktop_male['2015-02']['3_payment_page'], '\t\t', funnel_month_desktop_male['2015-03']['3_payment_page'], '\t\t', funnel_month_desktop_male['2015-04']['3_payment_page'])
    print('4_payment_confirmation_page\t', funnel_month_desktop_male['2015-01']['4_payment_confirmation_page'], '\t\t', funnel_month_desktop_male['2015-02']['4_payment_confirmation_page'], '\t\t', funnel_month_desktop_male['2015-03']['4_payment_confirmation_page'], '\t\t', funnel_month_desktop_male['2015-04']['4_payment_confirmation_page'])
    print('\n')



Таблица 3	 Воронка, пользователь: Мужчина, устройство: ПК, кол-во переходов
Страница/Месяц			 2015-01	 2015-02	 2015-03	 2015-04
1_home_page			 7504 		 7602 		 7538 		 7559
2_search_page			 3678 		 3832 		 3663 		 3836
3_payment_page			 596 		 586 		 134 		 164
4_payment_confirmation_page	 27 		 28 		 8 		 13




### 1.4 Пользователь: Женщина, устройство: ПК

In [5]:
funnel_month_desktop_female = {}
funnel_template = {'1_home_page': 0, '2_search_page': 0, '3_payment_page': 0, '4_payment_confirmation_page': 0}
with open('click_stream3.csv', mode='r') as csv_file: # открываем файл
    csv_reader = csv.DictReader(csv_file, fieldnames=['ID','page','date', 'device', 'Gender']) # читаем файл
    
    for row in csv_reader:
        
        page = list(row.items())[1][1]
        device = list(row.items())[3][1]
        Gender = list(row.items())[4][1]
        event_date = list(row.items())[2][1][:-3]
        
        if event_date not in funnel_month_desktop_female:
            funnel_month_desktop_female[event_date] = funnel_template.copy()
            
        if device == 'Desktop' and Gender == 'Female':
            
            
            if page == '1_home_page':
                funnel_month_desktop_female[event_date]['1_home_page'] += 1
            elif page == '2_search_page':
                funnel_month_desktop_female[event_date]['2_search_page'] += 1
            elif page == '3_payment_page':
                funnel_month_desktop_female[event_date]['3_payment_page'] += 1
            elif page == '4_payment_confirmation_page':
                funnel_month_desktop_female[event_date]['4_payment_confirmation_page'] += 1

    print('Таблица 4\t', 'Воронка, пользователь: Женщина, устройство: ПК, кол-во переходов')
    print('Страница/Месяц\t\t\t', '2015-01\t', '2015-02\t', '2015-03\t', '2015-04')
    print('1_home_page\t\t\t', funnel_month_desktop_female['2015-01']['1_home_page'], '\t\t', funnel_month_desktop_female['2015-02']['1_home_page'], '\t\t', funnel_month_desktop_female['2015-03']['1_home_page'], '\t\t', funnel_month_desktop_female['2015-04']['1_home_page'])
    print('2_search_page\t\t\t', funnel_month_desktop_female['2015-01']['2_search_page'], '\t\t', funnel_month_desktop_female['2015-02']['2_search_page'], '\t\t', funnel_month_desktop_female['2015-03']['2_search_page'], '\t\t', funnel_month_desktop_female['2015-04']['2_search_page'])
    print('3_payment_page\t\t\t', funnel_month_desktop_female['2015-01']['3_payment_page'], '\t\t', funnel_month_desktop_female['2015-02']['3_payment_page'], '\t\t', funnel_month_desktop_female['2015-03']['3_payment_page'], '\t\t', funnel_month_desktop_female['2015-04']['3_payment_page'])
    print('4_payment_confirmation_page\t', funnel_month_desktop_female['2015-01']['4_payment_confirmation_page'], '\t\t', funnel_month_desktop_female['2015-02']['4_payment_confirmation_page'], '\t\t', funnel_month_desktop_female['2015-03']['4_payment_confirmation_page'], '\t\t', funnel_month_desktop_female['2015-04']['4_payment_confirmation_page'])
    print('\n')
    

Таблица 4	 Воронка, пользователь: Женщина, устройство: ПК, кол-во переходов
Страница/Месяц			 2015-01	 2015-02	 2015-03	 2015-04
1_home_page			 7546 		 7448 		 7512 		 7491
2_search_page			 3851 		 3800 		 3707 		 3733
3_payment_page			 625 		 605 		 162 		 138
4_payment_confirmation_page	 33 		 26 		 6 		 9




## 2 Переход от кол-ва переходов к процентам относительно  1_home_page

In [6]:
funnel_month_mobile_male_percent = funnel_month_mobile_male.copy()
funnel_month_mobile_female_percent = funnel_month_mobile_female.copy()
funnel_month_desktop_male_percent = funnel_month_desktop_male.copy()
funnel_month_desktop_female_percent = funnel_month_desktop_female.copy()

In [7]:
for month in funnel_month_mobile_male_percent:
    funnel_month_mobile_male_percent[month]['4_payment_confirmation_page'] = str(float("{0:.2f}".format((funnel_month_mobile_male_percent[month]['4_payment_confirmation_page'] / funnel_month_mobile_male_percent[month]['1_home_page'])* 100)))+'%'
    funnel_month_mobile_male_percent[month]['3_payment_page'] = str(float("{0:.2f}".format((funnel_month_mobile_male_percent[month]['3_payment_page'] / funnel_month_mobile_male_percent[month]['1_home_page'])* 100)))+'%'
    funnel_month_mobile_male_percent[month]['2_search_page'] = str(float("{0:.2f}".format((funnel_month_mobile_male_percent[month]['2_search_page'] / funnel_month_mobile_male_percent[month]['1_home_page'])* 100)))+'%'
    funnel_month_mobile_male_percent[month]['1_home_page'] = '100%'
print('Таблица 5\t', 'Воронка, пользователь: Мужчина, устройство: Мобильный телефон, процент переходов с 1_home_page')
print('Страница/Месяц\t\t\t', '2015-01\t', '2015-02\t', '2015-03\t', '2015-04')
print('1_home_page\t\t\t', funnel_month_mobile_male_percent['2015-01']['1_home_page'], '\t\t', funnel_month_mobile_male_percent['2015-02']['1_home_page'], '\t\t', funnel_month_mobile_male_percent['2015-03']['1_home_page'], '\t\t', funnel_month_mobile_male_percent['2015-04']['1_home_page'])
print('2_search_page\t\t\t', funnel_month_mobile_male_percent['2015-01']['2_search_page'], '\t', funnel_month_mobile_male_percent['2015-02']['2_search_page'], '\t', funnel_month_mobile_male_percent['2015-03']['2_search_page'], '\t', funnel_month_mobile_male_percent['2015-04']['2_search_page'])
print('3_payment_page\t\t\t', funnel_month_mobile_male_percent['2015-01']['3_payment_page'], '\t', funnel_month_mobile_male_percent['2015-02']['3_payment_page'], '\t', funnel_month_mobile_male_percent['2015-03']['3_payment_page'], '\t\t', funnel_month_mobile_male_percent['2015-04']['3_payment_page'])
print('4_payment_confirmation_page\t', funnel_month_mobile_male_percent['2015-01']['4_payment_confirmation_page'], '\t\t', funnel_month_mobile_male_percent['2015-02']['4_payment_confirmation_page'], '\t\t', funnel_month_mobile_male_percent['2015-03']['4_payment_confirmation_page'], '\t\t', funnel_month_mobile_male_percent['2015-04']['4_payment_confirmation_page'])
print('\n')

for month in funnel_month_mobile_female_percent:
    funnel_month_mobile_female_percent[month]['4_payment_confirmation_page'] = str(float("{0:.2f}".format((funnel_month_mobile_female_percent[month]['4_payment_confirmation_page'] / funnel_month_mobile_female_percent[month]['1_home_page'])* 100)))+'%'
    funnel_month_mobile_female_percent[month]['3_payment_page'] = str(float("{0:.2f}".format((funnel_month_mobile_female_percent[month]['3_payment_page'] / funnel_month_mobile_female_percent[month]['1_home_page'])* 100)))+'%'
    funnel_month_mobile_female_percent[month]['2_search_page'] = str(float("{0:.2f}".format((funnel_month_mobile_female_percent[month]['2_search_page'] / funnel_month_mobile_female_percent[month]['1_home_page'])* 100)))+'%'        
    funnel_month_mobile_female_percent[month]['1_home_page'] = '100%'


print('Таблица 6\t', 'Воронка, пользователь: Женщина, устройство: Мобильный телефон, процент переходов с 1_home_page')
print('Страница/Месяц\t\t\t', '2015-01\t', '2015-02\t', '2015-03\t', '2015-04')
print('1_home_page\t\t\t', funnel_month_mobile_female_percent['2015-01']['1_home_page'], '\t\t', funnel_month_mobile_female_percent['2015-02']['1_home_page'], '\t\t', funnel_month_mobile_female_percent['2015-03']['1_home_page'], '\t\t', funnel_month_mobile_female_percent['2015-04']['1_home_page'])
print('2_search_page\t\t\t', funnel_month_mobile_female_percent['2015-01']['2_search_page'], '\t', funnel_month_mobile_female_percent['2015-02']['2_search_page'], '\t', funnel_month_mobile_female_percent['2015-03']['2_search_page'], '\t', funnel_month_mobile_female_percent['2015-04']['2_search_page'])
print('3_payment_page\t\t\t', funnel_month_mobile_female_percent['2015-01']['3_payment_page'], '\t', funnel_month_mobile_female_percent['2015-02']['3_payment_page'], '\t\t', funnel_month_mobile_female_percent['2015-03']['3_payment_page'], '\t\t', funnel_month_mobile_female_percent['2015-04']['3_payment_page'])
print('4_payment_confirmation_page\t', funnel_month_mobile_female_percent['2015-01']['4_payment_confirmation_page'], '\t\t', funnel_month_mobile_female_percent['2015-02']['4_payment_confirmation_page'], '\t\t', funnel_month_mobile_female_percent['2015-03']['4_payment_confirmation_page'], '\t\t', funnel_month_mobile_female_percent['2015-04']['4_payment_confirmation_page'])
print('\n')

for month in funnel_month_desktop_male_percent:
    funnel_month_desktop_male_percent[month]['4_payment_confirmation_page'] = str(float("{0:.2f}".format((funnel_month_desktop_male_percent[month]['4_payment_confirmation_page'] / funnel_month_desktop_male_percent[month]['1_home_page'])* 100)))+'%'
    funnel_month_desktop_male_percent[month]['3_payment_page'] = str(float("{0:.2f}".format((funnel_month_desktop_male_percent[month]['3_payment_page'] / funnel_month_desktop_male_percent[month]['1_home_page'])* 100)))+'%'
    funnel_month_desktop_male_percent[month]['2_search_page'] = str(float("{0:.2f}".format((funnel_month_desktop_male_percent[month]['2_search_page'] / funnel_month_desktop_male_percent[month]['1_home_page'])* 100)))+'%'        
    funnel_month_desktop_male_percent[month]['1_home_page'] = '100%'


print('Таблица 7\t', 'Воронка, пользователь: Мужчина, устройство: ПК, процент переходов с 1_home_page')
print('Страница/Месяц\t\t\t', '2015-01\t', '2015-02\t', '2015-03\t', '2015-04')
print('1_home_page\t\t\t', funnel_month_desktop_male_percent['2015-01']['1_home_page'], '\t\t', funnel_month_desktop_male_percent['2015-02']['1_home_page'], '\t\t', funnel_month_desktop_male_percent['2015-03']['1_home_page'], '\t\t', funnel_month_desktop_male_percent['2015-04']['1_home_page'])
print('2_search_page\t\t\t', funnel_month_desktop_male_percent['2015-01']['2_search_page'], '\t', funnel_month_desktop_male_percent['2015-02']['2_search_page'], '\t', funnel_month_desktop_male_percent['2015-03']['2_search_page'], '\t', funnel_month_desktop_male_percent['2015-04']['2_search_page'])
print('3_payment_page\t\t\t', funnel_month_desktop_male_percent['2015-01']['3_payment_page'], '\t\t', funnel_month_desktop_male_percent['2015-02']['3_payment_page'], '\t\t', funnel_month_desktop_male_percent['2015-03']['3_payment_page'], '\t\t', funnel_month_desktop_male_percent['2015-04']['3_payment_page'])
print('4_payment_confirmation_page\t', funnel_month_desktop_male_percent['2015-01']['4_payment_confirmation_page'], '\t\t', funnel_month_desktop_male_percent['2015-02']['4_payment_confirmation_page'], '\t\t', funnel_month_desktop_male_percent['2015-03']['4_payment_confirmation_page'], '\t\t', funnel_month_desktop_male_percent['2015-04']['4_payment_confirmation_page'])
print('\n')

for month in funnel_month_desktop_female_percent:
    funnel_month_desktop_female[month]['4_payment_confirmation_page'] = str(float("{0:.2f}".format((funnel_month_desktop_female[month]['4_payment_confirmation_page'] / funnel_month_desktop_female[month]['1_home_page'])* 100)))+'%'
    funnel_month_desktop_female[month]['3_payment_page'] = str(float("{0:.2f}".format((funnel_month_desktop_female[month]['3_payment_page'] / funnel_month_desktop_female[month]['1_home_page'])* 100)))+'%'
    funnel_month_desktop_female[month]['2_search_page'] = str(float("{0:.2f}".format((funnel_month_desktop_female[month]['2_search_page'] / funnel_month_desktop_female[month]['1_home_page'])* 100)))+'%'        
    funnel_month_desktop_female[month]['1_home_page'] = '100%'


print('Таблица 8\t', 'Воронка, пользователь: Женщина, устройство: ПК, процент переходов с 1_home_page')
print('Страница/Месяц\t\t\t', '2015-01\t', '2015-02\t', '2015-03\t', '2015-04')
print('1_home_page\t\t\t', funnel_month_desktop_female['2015-01']['1_home_page'], '\t\t', funnel_month_desktop_female['2015-02']['1_home_page'], '\t\t', funnel_month_desktop_female['2015-03']['1_home_page'], '\t\t', funnel_month_desktop_female['2015-04']['1_home_page'])
print('2_search_page\t\t\t', funnel_month_desktop_female['2015-01']['2_search_page'], '\t', funnel_month_desktop_female['2015-02']['2_search_page'], '\t', funnel_month_desktop_female['2015-03']['2_search_page'], '\t', funnel_month_desktop_female['2015-04']['2_search_page'])
print('3_payment_page\t\t\t', funnel_month_desktop_female['2015-01']['3_payment_page'], '\t\t', funnel_month_desktop_female['2015-02']['3_payment_page'], '\t\t', funnel_month_desktop_female['2015-03']['3_payment_page'], '\t\t', funnel_month_desktop_female['2015-04']['3_payment_page'])
print('4_payment_confirmation_page\t', funnel_month_desktop_female['2015-01']['4_payment_confirmation_page'], '\t\t', funnel_month_desktop_female['2015-02']['4_payment_confirmation_page'], '\t\t', funnel_month_desktop_female['2015-03']['4_payment_confirmation_page'], '\t\t', funnel_month_desktop_female['2015-04']['4_payment_confirmation_page'])


Таблица 5	 Воронка, пользователь: Мужчина, устройство: Мобильный телефон, процент переходов с 1_home_page
Страница/Месяц			 2015-01	 2015-02	 2015-03	 2015-04
1_home_page			 100% 		 100% 		 100% 		 100%
2_search_page			 80.47% 	 79.89% 	 19.99% 	 19.72%
3_payment_page			 14.96% 	 15.56% 	 4.2% 		 3.87%
4_payment_confirmation_page	 1.44% 		 1.55% 		 0.26% 		 0.34%


Таблица 6	 Воронка, пользователь: Женщина, устройство: Мобильный телефон, процент переходов с 1_home_page
Страница/Месяц			 2015-01	 2015-02	 2015-03	 2015-04
1_home_page			 100% 		 100% 		 100% 		 100%
2_search_page			 79.16% 	 80.51% 	 19.98% 	 20.32%
3_payment_page			 15.98% 	 16.8% 		 4.68% 		 3.95%
4_payment_confirmation_page	 1.96% 		 1.6% 		 0.54% 		 0.3%


Таблица 7	 Воронка, пользователь: Мужчина, устройство: ПК, процент переходов с 1_home_page
Страница/Месяц			 2015-01	 2015-02	 2015-03	 2015-04
1_home_page			 100% 		 100% 		 100% 		 100%
2_search_page			 49.01% 	 50.41% 	 48.59% 	 50.75%
3_payment_page			 7.94% 		

## 3 Анализ результатов

### 3.1 2_search_page
<p>По таблицам 5-6 видно, что произошло снижение процента переходов с 1_home_page на 2_search_page между февралем и мартом для мобильных устройств: для пользователей мужчин с 79.89% до 19.99%, для пользователей женщин с 80.51% до 19.98%.
По таблицам 7-8 особых изменией в доле переходов с 1_home_page на 2_search_page не наблюдается.</p>


### 3.2 3_payment_page
<p>По данным таблицам 5-6 можно заметить снижение процента пользователей дошедших до страницы 3_payment_page между февралем и мартом для мобильных устройств: для пользователей мужчин с 15.56% до 4.2%, для пользователей женщин с 16.8%% до 4.68%.
По данным таблиц 7-8 для пользователей ПК также произошло снижение: для пользователей мужчин с 15.56% до 4.2%, для пользователей женщин с 16.8%% до 4.68%.</p>

### 3.3 4_payment_confirmation_page
#### 3.3.1 Пользователи мобильных устройств
<p>В январе и феврале самые высокие показатели конверсияи наблюдается у пользователей мобильных устройств, причем у пользователей женского пола эти показатели немного выше. В марте и апреле также наблюдается большие показатели конверсии у пользователей мобильных устройств, причем в марте конверсия выше у пользователей-женщин, а в апреле у пользователей-мужчин.
Также следует отметить, что у пользоватетелй-женщин наблюдается динамика по снижению конверсии как в январе-феврале, так и в марте-апреле, у пользователей-мужчин, наоборот, наблюдается динамика на увеличение конверсии (таблицы 5-6). </p>

#### 3.3.2 Пользователи ПК
<p>Как описано выше, по сравнению в пользователями мобильных устройств, показатели конверсии у пользователей ПК ниже. Пользователи-женщины лидируют по уровню конверсии только в январе, дале показатели кнверсии выше у пользователей-мужчин. У пользователей мужчин в январе-феврале и в марте-апреле наблюдается динамика на увеличение показателя конверсии. У пользователей-женщин в январе-феврале наблюдается динамика на увеличение показателя конверсии, а в марте-апреле - динамика на его снижение (таблицы 7-8).</p>

## Выводы
<p>В данной работе произошло знакомство с воронкой продаж и произведен анализ динамики ее изменения с течением времени относительно пола пользователя и типа устройства.</p>